src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelNotificationBuilderTest.java

```diff
@@ -200,6 +248,116 @@ public void testNoActionButtons() {
         verify(builderSpy, never()).addAction(any(Notification.Action.class));
     }
 
+    public void testValidNotificationBadge() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_bdgcnt", 2);
+        mpPushSpy.createNotification(intent);
+        verify(builderSpy).setNumber(2);
+    }
+
+    public void testInvalidNotificationBadge() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_bdgcnt", "0");
+        mpPushSpy.createNotification(intent);
+        verify(builderSpy, never()).setNumber(any(Integer.class));
+    }
+
+    public void testChannelId() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_channel_id", "12345");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy).setChannelId("12345");
+        assertEquals(notification.getChannelId(), "12345");
+    }
+
+    public void testNoChannelId() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy).setChannelId(anyString());
+        assertEquals(notification.getChannelId(), MixpanelPushNotification.NotificationData.DEFAULT_CHANNEL_ID);
+    }
+
+    public void testSubText() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_subtxt", "SUBTEXT");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy).setSubText("SUBTEXT");
+    }
+
+    public void testNoSubText() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        Notification notification = mpPushSpy.createNotification(intent);
+        verify(builderSpy, never()).setSubText("MESSAGE");
```

Comment:
This test would still pass if `setSubText` is called with anything other than `"Message"`. This should be verifying the it's not called with any thing.